File: working/recursividad/21-fib/py/iterativo.py
class Entero:
    def __init__(self,n):
        self.__n = n

    def fibonacci(self):
        if self.__n == 0 or self.__n == 1:
            result = self.__n
        else:
            x = 0
            y = 1
            z = 0
            for i in range(2,self.__n+1):
                z = x + y
                x = y
                y = z
            result = z
        return result


print(Entero(6).fibonacci())

File: working/recursividad/21-fib/rb/iterativo.rb
class Entero
    def initialize(n)
        @n = n
    end

    def fibonacci
        if @n == 0 || @n == 1
            result = @n
        else
            x = 0
            y = 1
            z = 0
            (2..@n).each do |i|
                z = x + y
                x = y
                y = z
            end
            result = z
        end
        return result
    end
end


puts Entero.new(6).fibonacci
